packages/viz/src/lib/errors/layer-error.ts

```diff
@@ -0,0 +1,30 @@
+import { CartoError } from '@carto/toolkit-core';
+
+/**
+ * Utility to build a CartoError related to Layer errors.
+ *
+ * @return {CartoError} A well formed object representing the error.
+ */
+
+/**
+ * CartoPopupError types:
+ * - [Error]
+ * - [Missing property]
+ * - [Invalid coordinate]
+ *
+ * @name CartoLayerError
+ * @memberof CartoError
+ * @api
+ */
+export class CartoLayerError extends CartoError {
+  constructor(message: string, type = layerErrorTypes.DEFAULT) {
+    super({ message, type });
+    this.name = 'CartoLayerError';
+  }
+}
+
+export const layerErrorTypes = {
+  DEFAULT: '[Error]',
+  DECK_MAP_NOT_INSTANTIATED: '[Deck.gl map is not instantiated]',
```

Comment:
To follow the same pattern that we have in other error classes, we can put here [Deck.gl Map not found].

The kind of error should not be DECK_MAP_NOT_INSTANTIATED because map can be instantiated but the layer knows nothing about the map, I think we should put DECK_MAP_NOT_FOUND.

Reply:
Done